lib/spi/types.ts

```diff
@@ -221,4 +223,21 @@ export class ClusterInfo {
         assert(result, `Required resource ${name} is missing.`);
         return result!;
     }
+
+    /**
+     * Update executionContext map
+     * @param addOn
+     * @param Values
+     */
+    public addExecutionContext(addOn: string, values: Values) {
+        this.executionContext.set(addOn, values);
+    }
+
+    /**
+    * Returns all execution context
+    * @returns executionContext: Map<string, Values>
+    */
+    public getAllExecutionContext(): Map<string, Values> {
```

Comment:
Let's rename to `getAddOnContexts`. We can also have a method (later) like `getAddOnContext(addOn)`